src/main/java/seedu/duke/models/schema/Student.java

```diff
@@ -203,4 +214,136 @@ public ModuleList getModulesPlanned() {
+            String currModuleCode = currentSemModuleList.get(i).getModuleCode();
+            ModuleWeekly currModule = new ModuleWeekly(currModuleCode);
+            currentSemesterModulesWeekly.add(currModule);
+            //       currentSemesterModulesWeekly.add(new ModuleWeekly(currentSemesterModules.getModuleByIndex(i).
+            //               getModuleCode()));
+        }
+        //   for (int i = 0; i < currentSemesterModulesWeekly.size(); i++) {
+        //       System.out.println(currentSemesterModulesWeekly.get(i).getModuleCode());
+        //   }
+    }
+
+    public void plannerCommand(Student student, String userInput) {
+        setCurrentSemesterModules();
+        setCurrentSemesterModulesWeekly();
+        String argument = userInput.substring(userInput.indexOf("planner") + 7).trim().toUpperCase();
+        if (argument.equals("SHOW")) {
+            if (!checkValidTime(student)) {
+                WeeklyScheduleView.printWeeklySchedule(currentSemesterModulesWeekly);
+            }
+        }
+    }
+
+
+    /*
+    int startIndexOfTutorialStartTime = userNewInput.indexOf("/tutorial");
+                        int startIndexOfLectureStartTime = userNewInput.indexOf("/lecture");
+                        int startIndexOfLabStartTime = userNewInput.indexOf("/lab");
+                 //       int startIndexOfDuration = userNewInput.indexOf("/duration");
+                        int startIndexOfDay = userNewInput.indexOf("/day");
+                        String lecture = userNewInput.substring(startIndexOfLectureStartTime + 8,
+                                startIndexOfTutorialStartTime);
+                 //       String duration = userNewInput.substring(startIndexOfDuration + 9, startIndexOfDay);
+                        String day = userNewInput.substring(startIndexOfDay + 4);
+                        String tutorial = userNewInput.substring(startIndexOfTutorialStartTime + 9, startIndexO
+                        fLabStartTime);
+                        String lab = userNewInput.substring(startIndexOfLabStartTime + 6, startIndexOfDay);
+                        for (int i = 0; i < currentSemesterModulesWeekly.size(); i++) {
+                            if (!currentSemesterModulesWeekly.get(i).getModuleCode().equals(argument)) {
+                                System.out.println("error in exists by code");
+                                continue;
+                            }
+                            currentSemesterModulesWeekly.get(i).setDay(day);
+                            currentSemesterModulesWeekly.get(i).setLectureTime(Integer.parseInt(lecture.trim()));
+                            currentSemesterModulesWeekly.get(i).setLabTime(Integer.parseInt(lab.trim()));
+                            currentSemesterModulesWeekly.get(i).setTutorialTime(Integer.parseInt(tutorial.trim()));
+                        }
+     */
+
+    public boolean checkValidTime(Student student) {
```

Comment:
so does checkValidTime return true or false if the time is valid? From the code implementation it seems that true is for invalid times. But could that cause confusion for a developer using this function?

Reply:
Removed this function as I realized the weekly schedule view would never be shown if one of the parameters (lecture, tutorial, lab) for the modules is not entered by the user. You're right, it returns true when invalid, I'll be more careful of naming conventions.